File: TelerikUIExamplesInSwift2.0/TelerikUIExamplesInSwift/StackedAreaChart.swift
//
//  StackedAreaChart.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class StackedAreaChart: ExampleViewController {
    
    let chart = TKChart()
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)

        self.addOption("Stacked") { self.reloadData() }
        self.addOption("Stack 100") { self.reloadData() }
        self.addOption("No Stacking") { self.reloadData() }
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.exampleBoundsWithInset
        chart.autoresizingMask = UIViewAutoresizing(rawValue:UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        reloadData()
    }
    
    func reloadData() {
        chart.removeAllData()
        
        var stackInfo: TKChartStackInfo?
        if(self.selectedOption == 0) {
            stackInfo = TKChartStackInfo(ID: 1, withStackMode: TKChartStackMode.Stack)
        }
        else if (self.selectedOption == 1)
        {
            stackInfo = TKChartStackInfo(ID: 1, withStackMode: TKChartStackMode.Stack100)
        }
        
        for _ in 0..<3 {
            var array = [TKChartDataPoint]()
            for i in 0..<8 {
                array.append(TKChartDataPoint(x: i+1, y: Int(arc4random() % (100))))
            }
            let series = TKChartAreaSeries(items: array)
            series.selectionMode = TKChartSeriesSelectionMode.Series
            series.stackInfo = stackInfo
            chart.addSeries(series)
        }

        chart.reloadData()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

File: TelerikUIExamplesInCSharp/Examples/Chart/StackedCharts/StackedAreaChart.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	[Register("StackedAreaChart")]
	public class StackedAreaChart: XamarinExampleViewController
	{
		TKChart chart;

		public override void ViewDidLoad ()
		{
			this.AddOption ("Stacked", reloadData);
			this.AddOption ("Stacked 100", reloadData);
			this.AddOption ("No Stacking", reloadData);

			base.ViewDidLoad ();

			chart = new TKChart (this.View.Bounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			this.reloadData ();
		}

		void reloadData()
		{
			chart.RemoveAllData ();

			TKChartStackInfo stackInfo = null;

			if (this.SelectedOption == 0) 
				stackInfo = new TKChartStackInfo (new NSNumber (1), TKChartStackMode.Stack);
			else if (this.SelectedOption == 1)
				stackInfo = new TKChartStackInfo (new NSNumber (1), TKChartStackMode.Stack100);

			Random r = new Random ();
			for (int i = 0; i < 3; i++) {
				List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
				for (int j = 0; j < 8; j++) {
					list.Add(new TKChartDataPoint(new NSNumber(j+1), new NSNumber(r.Next() % 100)));
				}
				TKChartAreaSeries series = new TKChartAreaSeries (list.ToArray());
				series.SelectionMode = TKChartSeriesSelectionMode.Series;
				series.StackInfo = stackInfo;
				chart.AddSeries (series);
			}
		}
	}
}

